File: parralel-prog/Lab3/F3.cs
using System;
using System.Threading;
using static lab3.Lab3;

namespace lab3
{
    public class Function3
    {
        public void Run()
        {
            lock (Lab3.locker)
            {
                // F3 3.14  T = (O + P) * (ML * MS)
                
                //Thread.Sleep(1000);
                Vector P = new Vector(N);
                Vector O = new Vector(N);
                Matrix ML = new Matrix(N);
                Matrix MS = new Matrix(N);
                Vector result;

                P.FillVectorWithNumber(1);
                O.FillVectorWithNumber(1);
                ML.FillMatrixWithNumber(1);
                MS.FillMatrixWithNumber(1);

                Console.WriteLine("Function 3 start");
                result = ML.multiply(MS).multiply(O.sum(P));
                Console.WriteLine("Function 3 result \n" + result.toString());
                Console.WriteLine("Function 3 end \n");
            }
        }
    }

}

File: parralel-prog/Lab5/F3.cpp
#include "F3.h"

F3::F3(int N) {
	this->N = N;
}

Matrix* F3::getResult() {
	return result;
}

DWORD WINAPI F3::startThread(void* param) {
	F3* This = (F3*)param;
	return This->run();
}

DWORD F3::run() {
	cout << "Task 3 start\n";
	Matrix *MP = new Matrix(N);
	Matrix *MR = new Matrix(N);
	Matrix *MS = new Matrix(N);
	result = MS->sum(MP->multiply(MR));
	std::cout << result->toString() << std::endl;
	cout << "Task 3 end\n";
	delete MP;
	delete MR;
	delete MS;
	return 0;
}